scripts/ai/get-document-comment-tags.ts

```diff
@@ -0,0 +1,166 @@
+  storageInputContainerName: "comment-tags",
+  description: `Uses tags specified in comments.`
+};
+
+console.log(`*** Starting to Compile Document Tags ***`);
+
+const startTime = Date.now();
+const documentInfo: Record<string, DocumentInfo> = {};
+
+const databaseURL = "https://collaborative-learning-ec215.firebaseio.com";
+
+// Fetch the service account key JSON file contents; must be in same folder as script
+const credential = admin.credential.cert('./serviceAccountKey.json');
+// Initialize the app with a service account, granting admin privileges
+admin.initializeApp({
+  credential,
+  databaseURL
+});
+
+const credentialTime = Date.now();
+
+const sourcePath = `${datasetPath}${sourceDirectory}`;
+
+// Get network info from portal file. This should have been created by download-documents.ts.
+function getNetworkInfo() {
+  const networkFile = `${sourcePath}/${networkFileName}`;
+  if (fs.existsSync(networkFile)) {
+    return JSON.parse(fs.readFileSync(networkFile, "utf8"));
+  }
+}
+const { portal, demo } = getNetworkInfo() ?? { portal: "learn.concord.org" };
+
+// Determine ids of relevant documents by looking at files in source directory
+const tagCounts: Record<string, number> = {};
+const documentTags: Record<string, string[]> = {};
+fs.readdirSync(sourcePath).forEach(file => {
+  if (file.startsWith("document") && file.endsWith(".txt")) {
+    const id = file.split("document")[1].split(".txt")[0];
+    documentTags[id] = [];
+  }
+});
+
+console.log(`***** Getting document tags *****`);
+const tagStartTime = Date.now();
+const includedDocumentIds = Object.keys(documentTags);
+const collectionUrl = getFirebaseBasePath(portal, demo);
+const documentCollection = admin.firestore().collection(collectionUrl);
+
+let documentsProcessed = 0;
+let commentsProcessed = 0;
+let tagsProcessed = 0;
+let tagsIncluded = 0;
+let emptyTags = 0;
+
+// Look through all documents
+for (let i = 0; i < includedDocumentIds.length; i += queryLimit) {
+  // Documents are retrieved from firestore in batches so we can use the where query, which has a limit of 10.
+  console.log(`--- Checking documents ${i}-${i+queryLimit}`);
+  const documentIdSubset = includedDocumentIds.slice(i, i + queryLimit);
+  const documentSnapshots = await documentCollection.where("key", "in", documentIdSubset).get();
+
+  const processDocument =
+    async (documentSnapshot: admin.firestore.QueryDocumentSnapshot<admin.firestore.DocumentData>) => {
+      const documentStartTime = Date.now();
+      const documentId = documentSnapshot.data().key;
+
+      // Get the document's comments from firestore
+      const commentsUrl = `${documentSnapshot.ref.path}/comments`;
+      const commentCollection = admin.firestore().collection(commentsUrl);
+      const commentSnapshots = await commentCollection.get();
+
+      // Process each comment in series
+      const processComment =
+        async (commentSnapshot: admin.firestore.QueryDocumentSnapshot<admin.firestore.DocumentData>) => {
```

Comment:
I don't think you need to change it, but this function doesn't seem to be doing anything asynchronous. So you could probably drop the async here. And the await on line 127.

Reply:
I removed these. I'm still not fully wrapping my head around all the asynchronous stuff so any advice in that department is very welcome.